EIPS/eip-7254.md

```diff
@@ -0,0 +1,174 @@
+---
+eip: 7254
+title: Token Revenue Sharing
+description: Revenue token is a token that shares rewards for holders.
+author: Quy Phan (@quyphandang), Quy Phan <quy.phan@cryptoviet.info>
+discussions-to: https://ethereum-magicians.org/t/token-revenue-sharing/14872
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-06-29
+---
+
+## Abstract
+
+Revenue Sharing is an extended version of the standard token. It proposes an additional payment method for token holders. 
+
+This standard includes updating rewards for holders and allowing token holders to withdraw rewards.
+
+
+## Specification
+
+### Methods
+
+#### name
+
+Returns the name of the token
+
+``` js
+function name() public view returns (string)
+```
+
+
+#### symbol
+
+Returns the symbol of the token.
+
+``` js
+function symbol() public view returns (string)
+```
+
+
+
+#### decimals
+
+Returns the number of decimals the token uses
+
+``` js
+function decimals() public view returns (uint8)
+```
+
+
+#### totalSupply
+
+Returns the total token supply.
+
+``` js
+function totalSupply() public view returns (uint256)
+```
+
+
+
+#### balanceOf
+
+Returns the account balance of another account with address `account`.
+
+``` js
+function balanceOf(address account) public view returns (uint256)
+```
```

Comment:
A lot of this is identical to ERC-20. I recommend extending it for compatibility with other EIPs.

Reply:
I have updated